### PR TITLE
Add the ability for LogoSuite to properly render PNGs

### PR DESCRIPTION
**Pull request by rfearing**
## Summary

- Adds the ability to add transparent PNGs, instead of only SVGs, as sources for logos in the LogoSuite component.
	- Sets `filter` values to be applied to ``s based on the color values assigned to the `fill` property of the `<svg>`s.

## List of notable changes:

- **Added** the ability to use transparent `.png` images as sources for the Logos in the LogoSuite
	- In turn, added CSS to target and style them.
	- Adds PNG version inside Storybook

## What should reviewers focus on?

Compare the two methods to validate it looks comparible.

## Steps to test:

There is no new functionality. Compare:

- Original (with SVGs): http://127.0.0.1:6006/?path=/story/components-logosuite--playground
- Update (with PNGs): http://127.0.0.1:6006/?path=/story/components-logosuite--with-png-logos

## Supporting resources (related issues, external links, etc):

- Closes https://github.com/github/primer/issues/3230

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

**Emphasis on light color mode:**
emphasis-on-light

**Emphasis on dark color mode:**
emphasis-on-dark

**Muted on light color mode:**
muted-on-light

**Muted on dark color mode:**
muted-on-dark

### CODE REVIEW

**Review by joshfarrant — COMMENTED** [review]
Hey @rfearing 👋

This looks good to me, although I'd appreciate a 👍 from @rezrah too before merging this in if you're happy to hold off until early next week?

I know we've avoided including logos in the library previously, but as these are only included as fixtures and aren't bundled into the actual released package I imagine this will be fine. We already have SVG company logos in there, after all!

You'll need to update the snapshots before this can be merged too. You can just apply the _update snapshots_ label and it should happen automatically.

File: packages/react/src/LogoSuite/LogoSuite.stories.tsx
Comment (joshfarrant) [inline packages/react/src/LogoSuite/LogoSuite.stories.tsx:152]: Sorry, only just spotted this but could you add this story to LogoSuite.features.stories.tsx instead please? All of the stories for component-specific features live in there 🙂

**Review by joshfarrant — CHANGES_REQUESTED** [review]

File: packages/react/src/LogoSuite/LogoSuite.stories.tsx
Comment (rfearing) [reply packages/react/src/LogoSuite/LogoSuite.stories.tsx]: <picture data-single-emoji=":facepalm:" title=":facepalm:">:facepalm:</picture> Good catch. Changed.

**Comment by rfearing** [comment]
@joshfarrant I'm totally cool to wait for @rezrah. One caveat is, however, that I'll be on parental leave for a few months starting next week. Are you okay with  taking care of merging this if it gets approved?

File: packages/design-tokens/src/tokens/functional/components/logosuite/colors.json
Comment (rezrah) [inline packages/design-tokens/src/tokens/functional/components/logosuite/colors.json:15]: Thinking this should either be called `--brand-LogoSuite-color-logo-filter` or be placed straight into the CSS file 🤔. It currently overlaps with the existing token name, which is still valid.

If put straight to code, we could override the original token for the `img` selector only and apply these styles directly into the stylesheet.

File: packages/react/src/LogoSuite/LogoSuite.features.stories.tsx
Comment (rezrah) [inline packages/react/src/LogoSuite/LogoSuite.features.stories.tsx:366]: Nit, but png is incidental so would recommend calling this story `WithImgLogos` or `WithRasterLogos`.

File: packages/react/src/fixtures/images/logos/shopify.png
Comment (rezrah) [inline packages/react/src/fixtures/images/logos/shopify.png:1]: Can we use the full color logo for these fixtures?

**Review by rezrah — COMMENTED** [review]
@rfearing this is great, thanks so much 💝 . 

Here's some general feedback, which doesn't fit inline but is required to get this over the line...

- Updated visual snapshots
- A changeset
- Updating documentation, which current indicates that `svg` is the only valid child.
 
  Screenshot 2024-06-02 at 15 28 26

File: packages/react/src/fixtures/images/logos/shopify.png
Comment (joshfarrant) [reply packages/react/src/fixtures/images/logos/shopify.png:1]: 0be8361a4483f3b8dfa0ce79e2f41f40306a9b77

File: packages/react/src/LogoSuite/LogoSuite.features.stories.tsx
Comment (joshfarrant) [reply packages/react/src/LogoSuite/LogoSuite.features.stories.tsx]: 2eda631fbc103f90b2ebee3d3cc5632b55154065

File: packages/design-tokens/src/tokens/functional/components/logosuite/colors.json
Comment (joshfarrant) [reply packages/design-tokens/src/tokens/functional/components/logosuite/colors.json:15]: Could you expand on this a little please?

Are you suggesting we update `colors.json` to be as follows:

```json
{
  "brand": {
    "LogoSuite": {
      "color": {
        "logo": {
          "muted": {
            "value": "var(--brand-color-text-muted)",
            "dark": "var(--brand-color-text-muted)"
          },
          "emphasis": {
            "value": "var(--brand-color-text-default)",
            "dark": "var(--brand-color-text-default)"
          },
          "filter": {
            "muted": {
              "value": "brightness(0) saturate(100%) invert(37%) sepia(7%) saturate(778%) hue-rotate(171deg) brightness(96%) contrast(91%)",
              "dark": "brightness(0) saturate(100%) invert(61%) sepia(8%) saturate(430%) hue-rotate(171deg) brightness(94%) contrast(92%)"
            },
            "emphasis": {
              "value": "brightness(0) saturate(100%) invert(12%) sepia(20%) saturate(450%) hue-rotate(172deg) brightness(100%) contrast(91%)",
              "dark": "brightness(0) saturate(100%) invert(91%) sepia(2%) saturate(2455%) hue-rotate(193deg) brightness(107%) contrast(98%)"
            }
          }
        }
      }
    }
  }
}
```

For the alternative you suggested, could you point me in the direction of the token it overlaps with please? I'm struggling to find it.

Thanks! 🙂

Comment (rezrah) [reply packages/design-tokens/src/tokens/functional/components/logosuite/colors.json:15]: > Are you suggesting we update colors.json to be as follows

Yep, exactly that, because I see filter as an implementation child of logo, not an alternative / sibling.

**Comment by joshfarrant** [comment]
@rezrah As per your suggestion I've applied the filter to all logos regardless of whether they're vector or raster images.

It lgtm, but I'd appreciate your eyes on it too 🙂

**Comment by rezrah** [comment]
@joshfarrant - thanks tor these updates. Could you try re-running VRT updates please? Seems like a regression has appeared in LogoSuite, but might be false positive. I can't see a deployment link to verify the change, could that be added in please?

**Comment by joshfarrant** [comment]
@rezrah Link to Storybook here, will take a look at the Snapshots too

https://primer-fae531e7f3-26139705.drafts.github.io/storybook/?path=/story/components-logosuite-features--with-raster-logos

**Review by rezrah — APPROVED** [review]
Nicee... this is looking great :shipit:

**Review by joshfarrant — APPROVED** [review]

**Pull request merged**
